Report generation aborting because of inconsistent data types while indexing a property

This was recently reported by 2 users when generating a report from a previously processed case:

```
java.lang.Exception: Worker-9 Error while processing /RM1512-21/vol_vol3/$Unalloc/Unalloc_388982_53099524096_128406360064-Frag1>>Carved-838994144.mp4 (281555bytes)
                at iped.engine.core.Worker.process([Worker.java:186](http://worker.java:186/))
                at [iped.engine.core.Worker.run](http://iped.engine.core.worker.run/)([Worker.java:265](http://worker.java:265/))
Caused by: java.lang.IllegalArgumentException: Inconsistency of field data structures across documents for field [video:audio:video:audio:xmpDM:trackNumber:number] of doc [1121]. doc values type: expected 'SORTED_SET', but it has 'SORTED_NUMERIC
```

There is a weird video:audio:video:audio: prefix put in the property name above.

--- a/iped-carvers/iped-carvers-impl/src/main/java/iped/carvers/custom/MOVCarver.java
+++ b/iped-carvers/iped-carvers-impl/src/main/java/iped/carvers/custom/MOVCarver.java
@@ -28,50 +28,44 @@ public class MOVCarver extends AbstractCarver {
     private int defaultMaxLength = 500000000;
 
     public MOVCarver() throws DecoderException {
-        carverTypes = new CarverType[5];
+        carverTypes = new CarverType[6];
 
         carverTypes[0] = new CarverType();
         carverTypes[0].addHeader("????ftypmp41");
-        carverTypes[0].addHeader("????ftypmp42");
+        carverTypes[0].addHeader("????ftypmp42????mp");
+        carverTypes[0].addHeader("????ftypmp42????MP");
         carverTypes[0].addHeader("????ftypmmp4");
         carverTypes[0].addHeader("????ftypMSNV");
         carverTypes[0].addHeader("????ftypFACE");
         carverTypes[0].addHeader("????ftypdash");
         carverTypes[0].addHeader("????ftypisom");
         carverTypes[0].setMimeType(MediaType.parse("video/mp4"));
-        carverTypes[0].setMaxLength(defaultMaxLength);
-        carverTypes[0].setMinLength(defaultMinLength);
-        carverTypes[0].setName("MOV");
 
         carverTypes[1] = new CarverType();
         carverTypes[1].addHeader("????ftypmjp2");
         carverTypes[1].setMimeType(MediaType.parse("video/mj2"));
-        carverTypes[1].setMaxLength(defaultMaxLength);
-        carverTypes[1].setMinLength(defaultMinLength);
-        carverTypes[1].setName("MOV");
 
         carverTypes[2] = new CarverType();
         carverTypes[2].addHeader("????ftypM4V");
         carverTypes[2].setMimeType(MediaType.parse("video/x-m4v"));
-        carverTypes[2].setMaxLength(defaultMaxLength);
-        carverTypes[2].setMinLength(defaultMinLength);
-        carverTypes[2].setName("MOV");
 
         carverTypes[3] = new CarverType();
         carverTypes[3].addHeader("????ftyp3g");
         carverTypes[3].setMimeType(MediaType.parse("video/3gpp"));
-        carverTypes[3].setMaxLength(defaultMaxLength);
-        carverTypes[3].setMinLength(defaultMinLength);
-        carverTypes[3].setName("MOV");
 
         carverTypes[4] = new CarverType();
         carverTypes[4].addHeader("????ftypqt\\20\\20");
         carverTypes[4].setMimeType(MediaType.parse("video/quicktime"));
-        carverTypes[4].setMaxLength(defaultMaxLength);
-        carverTypes[4].setMinLength(defaultMinLength);
-        carverTypes[4].setName("MOV");
+
+        carverTypes[5] = new CarverType();
+        carverTypes[5].addHeader("????ftypmp42????M4A");
+        carverTypes[5].addHeader("????ftypmp42????m4a");
+        carverTypes[5].setMimeType(MediaType.parse("audio/mp4"));
 
         for (int i = 0; i < carverTypes.length; i++) {
+            carverTypes[i].setMaxLength(defaultMaxLength);
+            carverTypes[i].setMinLength(defaultMinLength);
+            carverTypes[i].setName("MOV");
             carverTypes[i].setCarverClass(this.getClass().getName());
         }
     }

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/util/MetadataUtil.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/util/MetadataUtil.java
@@ -392,10 +392,13 @@ public class MetadataUtil {
         normalizeGPSMeta(metadata);
         normalizeCase(metadata);
         prefixCommonMetadata(metadata);
-        prefixAudioMetadata(metadata);
-        prefixImageMetadata(metadata);
-        prefixVideoMetadata(metadata);
-        prefixPDFMetadata(metadata);
+        if (!prefixVideoMetadata(metadata)) {
+            if (!prefixAudioMetadata(metadata)) {
+                if (!prefixImageMetadata(metadata)) {
+                    prefixPDFMetadata(metadata);
+                }
+            }
+        }
         prefixDocMetadata(metadata);
         prefixBasicMetadata(metadata);
         removeDuplicateValues(metadata);
@@ -654,14 +657,20 @@ public class MetadataUtil {
         }
     }
 
-    private static void prefixAudioMetadata(Metadata metadata) {
-        if (metadata.get(Metadata.CONTENT_TYPE).startsWith("audio")) //$NON-NLS-1$
+    private static boolean prefixAudioMetadata(Metadata metadata) {
+        if (metadata.get(Metadata.CONTENT_TYPE).startsWith("audio")) {
             includePrefix(metadata, ExtraProperties.AUDIO_META_PREFIX);
+            return true;
+        }
+        return false;
     }
 
-    private static void prefixImageMetadata(Metadata metadata) {
-        if (metadata.get(Metadata.CONTENT_TYPE).startsWith("image")) //$NON-NLS-1$
+    private static boolean prefixImageMetadata(Metadata metadata) {
+        if (metadata.get(Metadata.CONTENT_TYPE).startsWith("image")) {
             includePrefix(metadata, ExtraProperties.IMAGE_META_PREFIX);
+            return true;
+        }
+        return false;
     }
 
     public static boolean isVideoType(MediaType mediaType) {
@@ -669,15 +678,21 @@ public class MetadataUtil {
                 || mediaType.getBaseType().toString().equals("application/vnd.rn-realmedia"); //$NON-NLS-1$
     }
 
-    private static void prefixVideoMetadata(Metadata metadata) {
+    private static boolean prefixVideoMetadata(Metadata metadata) {
         if (isVideoType(MediaType.parse(metadata.get(Metadata.CONTENT_TYPE)))
-                || isVideoType(MediaType.parse(metadata.get(StandardParser.INDEXER_CONTENT_TYPE))))
+                || isVideoType(MediaType.parse(metadata.get(StandardParser.INDEXER_CONTENT_TYPE)))) {
             includePrefix(metadata, ExtraProperties.VIDEO_META_PREFIX);
+            return true;
+        }
+        return false;
     }
 
-    private static void prefixPDFMetadata(Metadata metadata) {
-        if (metadata.get(Metadata.CONTENT_TYPE).equals("application/pdf")) //$NON-NLS-1$
+    private static boolean prefixPDFMetadata(Metadata metadata) {
+        if (metadata.get(Metadata.CONTENT_TYPE).equals("application/pdf")) {
             includePrefix(metadata, ExtraProperties.PDF_META_PREFIX);
+            return true;
+        }
+        return false;
     }
 
     private static void prefixDocMetadata(Metadata metadata) {

